pkg/admission/server.go

```diff
@@ -88,9 +89,18 @@ func (h *Server) Start(kubeconfigFlag string) error {
 	server := &http.Server{
 		Addr:    h.Address,
 		Handler: mux,
+		TLSConfig: &tls.Config{
```

Comment:
have you tried to benchmark the performance difference with this way? i am sure its not dramatic, but you have some numbers?

Reply:
running benchmark using [go-wrk](https://github.com/tsliwowicz/go-wrk) on [net-http-server](https://github.com/kmdrn7/go-by-example/tree/master/net-http-server) example project to test the performance between static and dynamic TLS

`go-wrk -no-vr -c 100 -d 60 https://localhost:8000`

==== DYNAMIC TLS ====
- attempt 1
```
Running 60s test @ https://localhost:8000
  100 goroutine(s) running concurrently
890582 requests in 59.416148154s, 49.26MB read
Requests/sec:           14988.89
Transfer/sec:           848.98KB
Avg Req Time:           6.671608ms
Fastest Request:        33.958µs
Slowest Request:        351.141203ms
Number of Errors:       0
```
- attempt 2
```
Running 60s test @ https://localhost:8000
  100 goroutine(s) running concurrently
921922 requests in 59.297414059s, 50.99MB read
Requests/sec:           15547.42
Transfer/sec:           880.62KB
Avg Req Time:           6.431933ms
Fastest Request:        33.293µs
Slowest Request:        323.69243ms
Number of Errors:       0
```
- attempt 3
```
Running 60s test @ https://localhost:8000
  100 goroutine(s) running concurrently
834767 requests in 59.421627632s, 46.17MB read
Requests/sec:           14048.20
Transfer/sec:           795.70KB
Avg Req Time:           7.118348ms
Fastest Request:        33.949µs
Slowest Request:        381.15829ms
Number of Errors:       0
```
==== STATIC TLS ====
- attempt 1
```
Running 60s test @ https://localhost:8000
  100 goroutine(s) running concurrently
704372 requests in 59.485841869s, 38.96MB read
Requests/sec:           11841.00
Transfer/sec:           670.68KB
Avg Req Time:           8.44523ms
Fastest Request:        33.69µs
Slowest Request:        305.516356ms
Number of Errors:       0
```
- attempt 2
```
Running 60s test @ https://localhost:8000
  100 goroutine(s) running concurrently
791154 requests in 59.529313549s, 43.76MB read
Requests/sec:           13290.16
Transfer/sec:           752.76KB
Avg Req Time:           7.524364ms
Fastest Request:        33.251µs
Slowest Request:        311.520355ms
Number of Errors:       0
```
- attempt 3
```
Running 60s test @ https://localhost:8000
  100 goroutine(s) running concurrently
945453 requests in 59.29277492s, 52.30MB read
Requests/sec:           15945.50
Transfer/sec:           903.16KB
Avg Req Time:           6.271361ms
Fastest Request:        34.728µs
Slowest Request:        352.149179ms
Number of Errors:       0
```

based on this test data, can we say that using this mechanism to load TLS is not causing a big performance issue?

Reply:
absolutely, the results look great. nice job!